File: sea_level_predictor.py
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from scipy.stats import linregress


def draw_plot():
    # Read data from file
    df = pd.read_csv("epa-sea-level.csv")
    extend_years1 = np.arange(1880, 2050, 1)
    extend_years2 = np.arange(2000, 2050, 1)

    # Create scatter plot
    fig, ax = plt.subplots()
    ax.scatter(x="Year", y="CSIRO Adjusted Sea Level", data=df)

    # Create first line of best fit
    best_fit1 = linregress(x=df["Year"], y=df["CSIRO Adjusted Sea Level"])
    plt.plot(
        extend_years1,
        best_fit1.intercept + best_fit1.slope * extend_years1,
        "r",
        label="Fitted line using all years",
    )
    # Create second line of best fit
    best_fit2 = linregress(
        x=df[df["Year"] >= 2000]["Year"],
        y=df[df["Year"] >= 2000]["CSIRO Adjusted Sea Level"],
    )
    plt.plot(
        extend_years2,
        best_fit2.intercept + best_fit2.slope * extend_years2,
        "y",
        label="Fitted line using only the years after 2000",
    )

    # Add labels and title
    ax.set_xlabel("Year")
    ax.set_ylabel("Sea Level (inches)")
    ax.set_title("Rise in Sea Level")
    ax.set_xticks(np.arange(1850.0, 2100.0, 25.0))
    plt.legend()

    # Save plot and return data for testing (DO NOT MODIFY)
    plt.savefig("sea_level_plot.png")
    return plt.gca()

File: sea-level-predictor.r
library(tidyverse)

df <- read_csv("epa-sea-level.csv")

# Create a scatterplot with x="Year" and y="CSIRO Adjusted Sea Level"
# Add the line of best fit, predicting up to year 2050
p <- df %>%
    ggplot(aes(x = Year, y = `CSIRO Adjusted Sea Level`)) +
    geom_point() +
    xlim(1880, 2050) +
    ylim(0.0, 15.0) +
    stat_smooth(
        method = "lm",
        size = 1,
        color = "red",
        se = FALSE,
        fullrange = TRUE
    )

# Recalculate the line of best fit using just data from 2000-2013
df_subset <- df[df$Year >= 2000, ]

fit_subset <- lm(`CSIRO Adjusted Sea Level` ~ Year, data = df_subset)

int <- coef(fit_subset)[1]
slope <- coef(fit_subset)[2]

# Add second best fit line to the scatter plot, predicting up to year 2050
p + geom_abline(slope = slope, intercept = int, color = "blue", size = 1) +
    labs(x = "Year", y = "Sea Level (inches)", title = "Rise in Sea Level") +
    theme(plot.title = element_text(hjust = 0.5)) +
    annotate(
        geom = "text",
        label = "Fitted line using only the years after 2000",
        x = 2025,
        y = 15,
        color = "blue"
    ) +
    annotate(
        geom = "text",
        label = "Fitted line using all years",
        x = 2045,
        y = 10.5,
        color = "red"
    )